core/dbt/constants.py

```diff
@@ -1,2 +1,3 @@
 SECRET_ENV_PREFIX = "DBT_ENV_SECRET_"
 DEFAULT_ENV_PLACEHOLDER = "DBT_DEFAULT_PLACEHOLDER"
+METADATA_ENV_PREFIX = "DBT_ENV_CUSTOM_ENV_"
```

Comment:
This name might be natural in terms of some context I don't have yet, but I wonder if it can be improved. Maybe DBT_LOGGING_VAR_ or DBT_METADATA_VAR_?

Reply:
This is actually an existing name ([docs](https://docs.getdbt.com/reference/artifacts/dbt-artifacts#:~:text=env%3A%20Any%20environment%20variables%20prefixed%20with%20DBT_ENV_CUSTOM_ENV_%20will%20be%20included%20in%20a%20dictionary%2C%20with%20the%20prefix%2Dstripped%20variable%20name%20as%20its%20key.)).  Metadata is in the manifest and these vars end up there currently.  I centralized the constant into this (newish) file as part of my changes.

Reply:
It would be my preference to have a new key for this behavior, such as DBT_LOG_CUSTOM_ENV.  Users already have existing DBT_ENV_CUSTOM_ENV environment variables, which they might actually be surprised to find in every log line. @jtcohen6 could you weigh in on this?

To me, this seems to be a different use case than we already have for DBT_ENV_CUSTOM_ENV.

Reply:
I'm not sure specifically how users are using `DBT_ENV_CUSTOM_ENV` currently.

I do see what you're saying with users possibly being surprised these env vars show up in logs now.  But is there a use case for these being the exact variables they want to show up?

Reply:
Right now those DBT_ENV_CUSTOM_ENV variables show up in the artifacts that are written up in the end, and there was a recent change to put them in the context for access in Jinja.

I'm not familiar with how users might use those, but from some of our test cases, it does look like they were intended for putting things like a run_id or a job_id. So maybe it would be fine to use this prefix string.

Reply:
Thanks for tagging me in, and for the good conversation here!

Users set these env vars with the explicit intent of having them show up in metadata artifacts. It makes it easier to debug and quickly cross-reference. I see logs as a metadata artifact, in very much the same vein.

This was also our intent by crafting such an unwieldy prefix. It's hard to type `DBT_ENV_CUSTOM_ENV_` by accident. I'm definitely open to a better name (`DBT_ENV_CUSTOM_METADATA_`?) that's equally clear about what it's doing. By convention, "functional" env vars should just be prefixed with `DBT_`, including env vars setting global configs.

I think it's an okay assumption to automatically coordinate between these. If later on, we hear a valid use case for having them show up in one, and not the other, I'm willing to admit that I was wrong! But I'd rather do this than require users to set two sets of identical env vars from the get-go.

Reply:
Thanks for the input @jtcohen6!  I do think `DBT_ENV_CUSTOM_METADATA_` does a better job of saying what the var is for sure, so lean towards this new name.

My understanding is that if we change the name we should add a deprecation warning for the old name and remove it in 2.0.  My question is would `DBT_ENV_CUSTOM_ENV_` only show up in the artifacts (since that's maintaining current behavior) and `DBT_ENV_CUSTOM_METADATA_` shows up in both logs and artifacts?

Reply:
I'd still vote in favor of coordinating the information so that it shows up in both places.

How would we feel about implementing this as `DBT_ENV_CUSTOM_ENV_` for now, and then opening a new ticket for the rename (+ deprecation warning)?